sencha-workspace/SlateDemonstrationsStudent/app/view/Dashboard.js

```diff
@@ -98,6 +98,18 @@ Ext.define('SlateDemonstrationsStudent.view.Dashboard', {
          */
         legend: false,
 
+        /**
+         * @cfg {Ext.Component|Object|boolean}
+         * Instance or configuration for container containing non-enrollment message.
+         */
+        nonEnrollmentMessage: {
+            xtype: 'component',
+            itemId: 'non_enrollment_message',
+            cls: 'slate-placeholder',
+            html: 'Not currently enrolled in this competency',
+            hidden: true
+        },
```

Comment:
If we're using a factory to build this instance and creating the `NonEnrollmentMessage` view class as referenced in the comment lower down, all this content should become part of that class and this config should just be defaulting to `false` (hidden) like the others

Also if we're moving this to a dedicated component with its own class name/xtype, it won't need `itemId` anymore

Reply:
@themightychris It felt a bit clunky to create a component class in its own file to display something that's similar to a "data not found" error.  But using an Ext.window.MessageBox seemed like overkill too.  I considered changing the text of the placeholderItem, but that felt hacky.  But it does make the Dashboard view cleaner: https://github.com/SlateFoundation/slate-cbl/pull/636/commits/bdb91989a90b2528573cbdd8eaf95bb778efced5  If there's a more Slate standard way to do this let me know.

Reply:
@BillClinton yeah great instinct. Instead of creating a new class, we could have also just had the `apply*` method fill in config for a base `Ext.Component` factory like this example does for a `TextArea`: https://github.com/SlateFoundation/slate-cbl/blob/develop/sencha-workspace/packages/slate-cbl/src/field/comments/Field.js#L50

But since you had already written up a name for the new class in the code I decided not to overcomplicate my feedback. I probably would have went this route instead of making the class in this case. The nice thing about having the class though is that sometimes we create a school-specific Sencha package that is just a bundle of class overrides, and having things broken out into named classes with copy/templates in config defaults makes it really easy for overrides to customize things